packages/frontend-core/src/api/rowActions.ts

```diff
@@ -1,50 +1,68 @@
-export const buildRowActionEndpoints = API => ({
+import {
+  RowActionsResponse,
+  RowActionResponse,
+  CreateRowActionRequest,
+  RowActionPermissionsResponse,
+  RowActionTriggerRequest,
+} from "@budibase/types"
+import { BaseAPIClient } from "./types"
+
+export interface RowActionEndpoints {
+  fetch: (tableId: string) => Promise<Record<string, RowActionResponse>>
+  create: (tableId: string, name: string) => Promise<RowActionResponse>
+  delete: (tableId: string, rowActionId: string) => Promise<void>
+  enableView: (
+    tableId: string,
+    rowActionId: string,
+    viewId: string
+  ) => Promise<RowActionPermissionsResponse>
+  disableView: (
+    tableId: string,
+    rowActionId: string,
+    viewId: string
+  ) => Promise<RowActionPermissionsResponse>
+  trigger: (
+    sourceId: string,
+    rowActionId: string,
+    rowId: string
+  ) => Promise<void>
+}
+
+export const buildRowActionEndpoints = (
+  API: BaseAPIClient
+): RowActionEndpoints => ({
   /**
    * Gets the available row actions for a table.
    * @param tableId the ID of the table
    */
   fetch: async tableId => {
-    const res = await API.get({
-      url: `/api/tables/${tableId}/actions`,
-    })
-    return res?.actions || {}
+    return (
+      await API.get<RowActionsResponse>({
+        url: `/api/tables/${tableId}/actions`,
+      })
+    ).actions
   },
 
   /**
    * Creates a row action.
    * @param name the name of the row action
    * @param tableId the ID of the table
    */
-  create: async ({ name, tableId }) => {
-    return await API.post({
+  create: async (tableId, name) => {
+    return await API.post<CreateRowActionRequest, RowActionResponse>({
       url: `/api/tables/${tableId}/actions`,
       body: {
         name,
       },
     })
   },
 
-  /**
-   * Updates a row action.
-   * @param name the new name of the row action
-   * @param tableId the ID of the table
-   * @param rowActionId the ID of the row action to update
-   */
-  update: async ({ tableId, rowActionId, name }) => {
```

Comment:
Was the update no longer needed?

Reply:
Yep this was also dead code! Row actions are now renamed by renaming the automation, using the normal automation endpoint, rather than via this one.